src/utils/transactionUtils/transactionUtils.ts

```diff
@@ -405,3 +415,28 @@ export function mapTransactionDataKeyToPayload(
     functionName: TRANSACTION_DATA_KEYS[interactionType].functionName,
   };
 }
+
+export function getLinkId(
+  interactionType: ValidInteractionType,
+  transactionData: TransactionData,
+): string {
+  if (
+    interactionType === INTERACTION_TYPES.BUY_RECORD &&
+    isObjectOfTransactionPayloadType<BuyRecordPayload>(
+      transactionData,
+      TRANSACTION_DATA_KEYS[INTERACTION_TYPES.BUY_RECORD].keys,
+    )
+  ) {
+    return transactionData.contractTxId.toString() ?? '';
+  }
+  if (
+    interactionType === INTERACTION_TYPES.CREATE &&
+    isObjectOfTransactionPayloadType<CreatePDNTPayload>(
+      transactionData,
+      TRANSACTION_DATA_KEYS[INTERACTION_TYPES.CREATE].keys,
+    )
+  ) {
+    return transactionData.deployedTransactionId?.toString() ?? '';
+  }
+  return transactionData.assetId.toString() ?? '';
```

Comment:
i don't think `assetId` can be null, so no null-coalesce needed here

Reply:
added